packages/jest-matcher-utils/src/index.ts

```diff
@@ -293,13 +293,6 @@ const isLineDiffable = (expected: unknown, received: unknown): boolean => {
     return false;
   }
 
-  if (
-    expectedType === 'object' &&
-    typeof (expected as any).asymmetricMatch === 'function'
-  ) {
-    return false;
-  }
-
   if (
```

Comment:
seems odd to me to remove the check for `expectedType`, but not `receivedType`? Although I guess passing an asymmetric matcher as `received` is weird, so maybe not